awx/main/models/projects.py

```diff
@@ -194,6 +194,11 @@ def get_project_path(self, check_if_exists=True):
             if not check_if_exists or os.path.exists(smart_str(proj_path)):
                 return proj_path
 
+    def get_cache_path(self):
+        local_path = os.path.basename(self.local_path)
+        if local_path:
```

Comment:
Should we put a `logger.warn` in an `else` here?

Reply:
Or perhaps a `.debug()`, `warn` might be too noisy.

Reply:
This is a mess. The only reason it's used is because this is the status-quo:

https://github.com/ansible/awx/blob/8b97e3f8bb168f77045ffb2f15cfffc8e77fa005/awx/main/models/projects.py#L191

If `local_path` is ever `None`, it will error in almost any case. We don't handle the null case. What would it even mean if you did get that? That would happen if `local_path` is `foo/`, but we auto-generate this field, and I can't figure out anyway the auto-generation would produce that. _Maybe_ manual projects could accept this kind of value. If it did, this wouldn't be right. You would prefer to `rstrip` the slash instead.

It looks like the code is, for some reason, expecting to be given an absolute path. I don't know why.